### ISSUE TITLE
Export Protocol and ConnectionLike

### ISSUE THREAD

**Issue by xNxExOx** [blackbeam/mysql_async#48]
I need these to be able to do:
```rs
extern crate mysql_async as my;
use my::{QueryResult, Protocol, ConnectionLike};
use my::prelude::FromRow;
use tokio_async_await::compat::forward::IntoAwaitable;

pub async fn get_all_results<TupleType, T, P>(result : QueryResult<T, P>) -> Result<Vec<TupleType>, my::error::Error>
    where TupleType: FromRow + Send + 'static,
          P: Protocol + Send + 'static,
          T: ConnectionLike + Sized + Send + 'static,  {
    let mut fut = result.collect();
    let (_, data) :(_, Vec<TupleType>) = match await!(fut.into_awaitable()) {
        Ok(r) => r,
        Err(e) => {return Err(e);}
    };
    Ok(data)
}

pub async fn get_single_result<TupleType, T, P>(result : QueryResult<T, P>) -> Result<TupleType, my::error::Error>
    where TupleType: FromRow + Send + 'static,
          P: Protocol + Send + 'static,
          T: ConnectionLike + Sized + Send + 'static,  {
    let mut data : Vec<TupleType> = await!(get_all_results(result))?;
    if data.len() != 1 {
        Err(my::error::Error::Io(std::io::Error::from(std::io::ErrorKind::InvalidData)))
    }else {
        Ok(data.remove(0))
    }
}```
right now I need to use modified version to to make them public exported.

**Comment by blackbeam** [comment]
Hi. Pipes/sockets support is in progress atm. I'll look into your issue after.

**Comment by xNxExOx** [comment]
I changed line 195 in lib.rs to:
`pub use self::queryable::{BinaryProtocol, TextProtocol, Protocol};`
and added new line `pub use self::connection_like::ConnectionLike;`

**Comment by xNxExOx** [comment]
Of course it would be even better if more functions like these would be available directly on the `QueryResult` :) but if my code will work with standard crate version it will be enough for now :)

**Comment by blackbeam** [comment]
Fixed in [v0.19.0](https://github.com/blackbeam/mysql_async/releases/tag/v0.19.0)
See also [this test](https://github.com/blackbeam/mysql_async/commit/65351e224146ab2074c7fbd6cb23382ff71c6a38#diff-20feb0362721900efe49097986c17425R68).

**Comment by xNxExOx** [comment]
thanks
